src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variants.html.twig

```diff
@@ -0,0 +1,138 @@
+                                {{ 'sylius.ui.shipping'|trans }}
+                            </div>
+                            {% if variant.shippingCategory is same as(null) and variant.width is same as(null) and variant.height is same as(null) and variant.depth is same as(null) and variant.weight is same as(null) %}
+                                <span class="ui teal label"><i class="remove icon"></i>{{ 'sylius.ui.no_shipping_data'|trans }}</span>
+                            {% else %}
+                            <table class="ui definition table">
+                                <tbody>
+                                <tr>
+                                    <td class="five wide">{{ 'sylius.ui.shipping_category'|trans }}</td>
+                                    <td>{{ variant.shippingCategory }}</td>
+                                </tr>
+                                <tr>
+                                    <td class="five wide">{{ 'sylius.ui.width'|trans }}</td>
+                                    <td>{{ variant.width }}</td>
+                                </tr>
+                                <tr>
+                                    <td class="five wide">{{ 'sylius.ui.height'|trans }}</td>
+                                    <td>{{ variant.height }}</td>
+                                </tr>
+                                <tr>
+                                    <td class="five wide">{{ 'sylius.ui.depth'|trans }}</td>
+                                    <td>{{ variant.depth }}</td>
+                                </tr>
+                                <tr>
+                                    <td class="five wide">{{ 'sylius.ui.weight'|trans }}</td>
+                                    <td>{{ variant.weight }}</td>
+                                </tr>
+                                </tbody>
+                            </table>
+                            {% endif %}
+                        </div>
+                        <div class="four wide column pricing">
+                            <div class="ui small header">
+                                {{ 'sylius.ui.pricing'|trans }}
+                            </div>
+                            <table class="ui definition table">
+                                <thead>
+                                <th>{{ 'sylius.ui.channels'|trans }}</th>
+                                <th>{{ 'sylius.ui.price'|trans }}</th>
+                                <th>{{ 'sylius.ui.original_price'|trans }}</th>
+                                </thead>
+                                <tbody>
+                                {% for channelPricing in variant.channelPricings %}
+                                    <tr>
+                                        <td class="five wide">{{ channelPricing.channelCode }}</td>
+                                        <td>{{ channelPricing.price|sylius_variant_price(channelPricing.channelCode) }}</td>
```

Comment:
Every call to this filter would result in a database query to find the channel by its code.

Reply:
do you have some idea, how do this without many queries? From ``ChannelPricing.php`` I can get only channel code, I don't have information about currency without finding it by channel code :(